src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,339 @@
+            if(isDone) {
+                event.markAsDone();
+            }
+            if(isNew) {
+                echoTask(event);
+            }
+        } catch (StringIndexOutOfBoundsException e) {
+            printError(ERROR_INVALID_INPUT);
+        }
+    }
+
+    public static void echoTask(Task task) {
+        printLine();
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + task.toString());
+        String wordTask = (arrTasks.size() > 1) ? " tasks" : " task";
+        System.out.println("Now you have " + arrTasks.size() + wordTask + " in the list.");
+        printLine();
+    }
+
+    public static void markAsDone(String inputNum) throws DukeException {
+        try {
+            int itemIndex = getIndexFromInput(inputNum,
+                    "Please input a number to mark task as done.");
+            if(arrTasks.get(itemIndex).getStatusIcon().equals("[\u2713]")) {
+                throw new DukeException(ERROR_MARKED_TASK);
+            }
+            arrTasks.get(itemIndex).markAsDone();
+            printDone(itemIndex);
+        } catch (NumberFormatException e) {
+            printError(ERROR_INVALID_TASK_NUMBER);
+        }
+    }
+
+    public static void printDone(int doneItemIndex) {
+        printLine();
+        System.out.println("Nice! I've marked this task as done: \n\t"
+                + arrTasks.get(doneItemIndex));
+        printLine();
+    }
+
+    public static String[] splitCommandAndTask(String args) {
+        final String[] line = args.trim().split(" ",2);
+        if (line.length == 2) {
+            return line;
+        }
+        else {
+            return new String[] {line[0], null};
+        }
+    }
+
+    public static int getIndexFromInput(String inputNum, String nullErrMessage) throws DukeException {
+        if(inputNum == null) {
+            throw new DukeException(nullErrMessage);
+        }
+        int itemIndex = Integer.parseInt(inputNum) - 1;
+        if (itemIndex < 0 || itemIndex >= arrTasks.size()) {
+            throw new DukeException(ERROR_INVALID_TASK_NUMBER);
+        }
+        return itemIndex;
+    }
+
+    public static void deleteTask(String inputNum) throws DukeException {
+        try {
+            int deletedItemIndex = getIndexFromInput(inputNum,
+                    "Please input a number to delete task.");
+            printLine();
+            System.out.println("Noted. I've removed this task:  \n\t"
+                    + arrTasks.get(deletedItemIndex));
+            arrTasks.remove(deletedItemIndex);
+            String wordTask = (arrTasks.size() > 1) ? " tasks" : " task";
+            System.out.println("Now you have " + arrTasks.size() + wordTask + " in the list.");
+            printLine();
+        } catch (DukeException err) {
+            throw err;
+        }
+    }
+
+    private static void writeFile(String filePath, String text) throws IOException {
+        FileWriter file = new FileWriter(filePath);
```

Comment:
The name of the variable is a bit misleading. it should be a "fileWriter" instead of a "file".